lua/indent_blankline/init.lua

```diff
@@ -160,10 +166,13 @@ local refresh = function()
         context_status, context_start, context_end = utils.get_current_context(v("indent_blankline_context_patterns"))
     end
 
+    local prev_indent = nil
+
     local get_virtual_text = function(indent, extra, blankline, context_active, context_indent, virtual_string)
         local virtual_text = {}
         local current_left_offset = left_offset
-        for i = 1, math.min(math.max(indent, 0), max_indent_level) do
+        local local_max_indent_level = math.min(max_indent_level, prev_indent + max_indent_increase)
```

Comment:
`get_virtual_text` can't use dynamic variables from outside. It runs async. This is not guaranteed to work.
Revert it back to how you had it before. Update the indent value before you call `get_virtual_text`

Reply:
Maybe I can just pass prev_indent as a parameter?

Reply:
that works for now

really it shouldn't be the responsibility of `get_virtual_text`, but a lot of stuff in there shouldn't. I have to clean up the code at some point.